lib/bolt/transport/orch/connection.rb

```diff
@@ -17,13 +17,16 @@ def self.get_key(opts)
         end
 
         def initialize(opts, plan_context, logger)
+          require 'addressable/uri'
+
           @logger = logger
           @key = self.class.get_key(opts)
-          client_keys = %w[service-url token-file cacert job-poll-interval job-poll-timeout]
-          client_opts = client_keys.each_with_object({}) do |k, acc|
-            acc[k] = opts[k] if opts.include?(k)
-          end
+          client_opts = opts.slice('token-file', 'cacert', 'job-poll-interval', 'job-poll-timeout')
+          uri = Addressable::URI.parse(opts['service-url'])
+          uri&.port ||= 8143
+          client_opts['service-url'] = uri.to_s
```

Comment:
```suggestion
          if opts.key?('service-url')
            uri = Addressable::URI.parse(opts['service-url'])
            uri&.port ||= 8143
            client_opts['service-url'] = uri.to_s
          end
```

This will result in `service-url` being set to an empty string (`nil.to_s == ""`) if it's not set or not parseable, resulting in a misleading error:

```
  '' is an invalid service-url
```

We could just error here if `service-url` is not set or is not parseable since it's a required option, but that seems less preferable.